### ISSUE TITLE
AMQP 2.1.4 fails on Windows with  [Errno 10042] An unknown, invalid, or unsupported option or level was specified in a getsockopt or setsockopt call

### ISSUE THREAD

**Issue by cyberMe** [celery/py-amqp#135]
Tested on Windows7 x64 and Windows8 x32 with Python 2.7.13, 2.7.8, 3.6.0. RabbitMq 3.6.5.

Stack trace:
```
Traceback (most recent call last):
  File ".\main.py", line 12, in <module>
    c = main()
  File ".\main.py", line 8, in main
    conn.connect()
  File "C:\Python27\lib\site-packages\amqp\connection.py", line 294, in connect
    self.transport.connect()
  File "C:\Python27\lib\site-packages\amqp\transport.py", line 122, in connect
    self.socket_settings, self.read_timeout, self.write_timeout,
  File "C:\Python27\lib\site-packages\amqp\transport.py", line 174, in _init_socket
    self._set_socket_options(socket_settings)
  File "C:\Python27\lib\site-packages\amqp\transport.py", line 204, in _set_socket_options
    self.sock.setsockopt(SOL_TCP, opt, val)
  File "C:\Python27\lib\socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
socket.error: [Errno 10042] An unknown, invalid, or unsupported option or level was specified in a getsockopt or setsockopt call
```

Sample code:
```python
import amqp

def main():
	port = 5672
	host = 'localhost'
	user = 'guest'
	conn = amqp.connection.Connection(host=host, port=port, username=user, password=user)
	conn.connect()
	return conn

if __name__ == '__main__':
	c = main()
	if c.connected:
		c.close()
	del c
```

**Comment by frol** [comment]
This seems to be resolved in #136.

**Comment by auvipy** [comment]
thanks frol
